src/absil/bytes.fs

```diff
@@ -295,77 +295,95 @@ type ByteMemory with
 
     member x.AsReadOnly() = ReadOnlyByteMemory x
 
-    /// Mono and .Net Framework/CoreCLR have different implementations of memory-mapped files.
-    /// The mono implementation _requires_ a mapName, but the CoreCLR implementation _requires_ a name not be specified.
-    /// This member papers over those changes while mono bug https://github.com/mono/mono/issues/10245 is still active,
-    /// or until the implementations are unified
-    static member private CreateReadWriteMemoryMappedFileSafe(length: int64) =
-        let name =
+    static member CreateMemoryMappedFile(bytes: ReadOnlyByteMemory) =
+        let _makeMMFile (bytes: ReadOnlyByteMemory) =
```

Comment:
I could have split the calls here into the specific IF blocks, but since the netfx-on-windows case is the same as the netcore case, I thought it would make the most sense from a maintainability point of view to extract out each path into separate functions and call them in the ifdefs/match below.